operations-stdlib/src/commonMain/kotlin/string/Trim.kt

```diff
@@ -0,0 +1,61 @@
+package string
+
+import operation.StandardLogicOperation
+import utils.asList
+import utils.secondOrNull
+import utils.thirdOrNull
+
+object Trim : StandardLogicOperation, StringUnwrapStrategy {
+    override fun evaluateLogic(expression: Any?, data: Any?): Any?  {
+        return when (val firstItem = expression.asList.firstOrNull()) {
+            is String -> firstItem.trimElements(' ', TrimMode.BothEnds)
+            is List<*> -> {
+                val text = firstItem.firstOrNull()
+                val sign = firstItem.secondOrNull() ?: " "
+                val mode = (firstItem.thirdOrNull() as? String).toTrimMode() ?: TrimMode.BothEnds
+                (sign as? String)?.let {
+                    val char: Char = it.single()
+                    text.trimElements(char, mode)
+                }
+            }
+            else -> null
+        }
+    }
+
+    private fun String?.toTrimMode() = when (this) {
+        "start" -> TrimMode.Start
+        "end" -> TrimMode.End
+        "bothEnds" -> TrimMode.BothEnds
+        else -> null
+    }
+
+    private fun Any?.trimElements(char: Char, mode: TrimMode) =
+        when (this) {
+            is String -> {
+                modeBasedTrim(
+                    mode = mode,
+                    start = { trimStart(char) },
+                    end = { trimEnd(char) },
+                    bothEnds = { trim(char) }
+                )
+            }
+            else -> null
```

Comment:
This result is not being included in tests -
 case of having list as first argument ->
`operatorName to listOf(listOf(listOf(" cat "), " ", "bothEnds"))` - of course it should fail

same goes for making call from Data, which is not tested either (it should work out of the box, but having single verifier in your code won't hurt). 

`            TestInput(
                expression = mapOf(
                    operatorName to listOf(listOf(listOf(" cat "), " ", "bothEnds"))
                ),
                result = Success("cat")
            ),
            TestInput(
                expression = mapOf(
                    operatorName to listOf(listOf(mapOf("var" to "key"), " ", "bothEnds"))
                ),
                data = mapOf("key" to listOf(" cat ", " cat ")),
                result = Success("cat")
            ),`

Reply:
👍 I have fixed it in the new commit